components/form.js

```diff
@@ -375,6 +386,11 @@ export function InputUserSuggest ({ label, groupClassName, ...props }) {
                 setOValue(v.name)
                 setSuggestions(INITIAL_SUGGESTIONS)
               }}
+              onBlur={(e) => {
+                if (!document.getElementsByClassName(groupClassName)[0].contains(e.relatedTarget)) {
+                  setSuggestions(INITIAL_SUGGESTIONS)
+                }
+              }}
```

Comment:
I can see that you need the `if` check in the input `onBlur` listener to be able to still select items. The `onClick` listener then seems to close the dropdown.

But do you need _this_ `onBlur` listener here? It seemed to work without this.

Also, can't this be solved in a more "react-y way"; without using `document.getElementsByClassName` For example with `useRef`? I didn't try yet though.

Reply:
The simplest way to think about it is that if the focus leaves the whole group (inclusive of the input box and the suggested items) then and only then do we want to hide the menu. The blur can be triggered by clicking or by forward/backward tabbing. So the blur handler hides the menu IF the new focused element is anything outside the group. And for that, all elements in the group that can get focus have to have the blur function.

I wanted to use useRef (commonly used for focus-handling use cases) but unfortunately useRef is no longer supported on function components it seems.